graylog2-web-interface/src/pages/SystemOverviewPage.jsx

```diff
@@ -14,53 +14,74 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
-import React from 'react';
+import * as React from 'react';
+import { useState } from 'react';
 
-import { DocumentTitle, IfPermitted } from 'components/common';
+import { DocumentTitle, IfPermitted, Spinner } from 'components/common';
 import { IndexerClusterHealth, IndexerFailuresComponent } from 'components/indexers';
 import { NotificationsList } from 'components/notifications';
 import { SystemJobsComponent } from 'components/systemjobs';
 import { SystemMessagesComponent } from 'components/systemmessages';
 import { TimesList } from 'components/times';
 import { GraylogClusterOverview } from 'components/cluster';
 import HideOnCloud from 'util/conditional/HideOnCloud';
+import CombinedProvider from 'injection/CombinedProvider';
+import usePluginEntities from 'views/logic/usePluginEntities';
 
-class SystemOverviewPage extends React.Component {
-  render() {
-    return (
-      <DocumentTitle title="System overview">
-        <span>
-          <IfPermitted permissions="notifications:read">
-            <NotificationsList />
-          </IfPermitted>
+const { EnterpriseActions } = CombinedProvider.get('Enterprise');
+
+const SystemOverviewPage = () => {
+  const [loadIndexerFailuresComponent, setLoadIndexerFailuresComponent] = useState(<Spinner text="Looking for Index Failures..." />);
+
+  const pluginSystemOverview = usePluginEntities('systemOverview');
+  const EnterpriseIndexerFailures = pluginSystemOverview?.[0]?.component;
+
+  const renderEnterpriseIndexerFailures = () => {
+    if (EnterpriseIndexerFailures) {
+      EnterpriseActions.getLicenseInfo().then((response) => {
```

Comment:
We should not call `EnterpriseActions.getLicenseInfo()` on each render, since this is firing a network request and performing a side effect. Also, this is having a big impact on the number of re-renders and network requests we do.

Here is a video of the Overview page on 4.1 and React dev tools' "highlight updates when components render" turned on:

https://user-images.githubusercontent.com/716185/129735508-631cfbe4-c530-4348-aa39-920ace751431.mp4


And here is of this PR and React dev tools' "highlight updates when components render" turned on:

https://user-images.githubusercontent.com/716185/129736015-ba91a41d-5ea7-4194-833c-11cfbf0beb21.mp4


I am not sure if the rebases I did on my local branches are causing this issue but I guess it's better to double check it.

Reply:
Thanks for spotting this @edmundoa 🎉 I have pushed up a resolution!

Reply:
This looks much better now, thank you for fixing it!